File: Java-Source/Day15.java
//DAY 15 OF HACKERRANK 30 DAY CHALLENGE: LINKED LIST (JAVA)

import java.util.*;

class Node {
    int data;
    TreeNode next;
    Node(int d) {
        data = d;
        next = null;
    }
}

public class Day15 {
    public static TreeNode insert(TreeNode head, int data) {
        TreeNode traverser = head;

        if (head == null) {
            head = new TreeNode(data);
        }
        else {
            while (traverser.next != null) {
                traverser = traverser.next;
            }
            traverser.next = new TreeNode(data);
        }

        return head;
    }
    public static void display(TreeNode head) {
        TreeNode start = head;
        while(start != null) {
            System.out.print(start.data + " ");
            start = start.next;
        }
    }

    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        TreeNode head = null;
        int N = sc.nextInt();

        while(N-- > 0) {
            int ele = sc.nextInt();
            head = insert(head,ele);
        }
        display(head);
        sc.close();
    }
}

File: CPP-Source/Day15.cpp
//DAY 15 OF HACKERRANK 30 DAY CHALLENGE: LINKED LIST (C++)

#include <iostream>
#include <cstddef>
using namespace std;	
class Node
{
    public:
        int data;
        Node *next;
        Node(int d){
            data=d;
            next=NULL;
        }
};
class Solution{
    public:  
      Node* insert(Node *head,int data)
      {
          if (head == NULL) {
              head = new Node(data);
          }
          else {
              Node *traverser = head;
              while (traverser->next != NULL) {
                  traverser = traverser->next;
              }
              traverser->next = new Node(data);
          }
          
          return head;
      }
      void display(Node *head)
      {
          Node *start=head;
          while(start)
          {
              cout<<start->data<<" ";
              start=start->next;
          }
      }
};
int main()
{
	Node* head=NULL;
  	Solution mylist;
    int T,data;
    cin>>T;
    while(T-->0){
        cin>>data;
        head=mylist.insert(head,data);
    }	
	mylist.display(head);
		
}